File: clients/PHP-client/src/ScriptsClient.php
<?php

namespace SoliDB;

class ScriptsClient
{
    private Client $client;

    public function __construct(Client $client)
    {
        $this->client = $client;
    }

    public function create(string $name, string $path, array $methods, string $code, ?string $description = null, ?string $collection = null): array
    {
        $params = [
            'database' => $this->client->getDatabase(),
            'name' => $name,
            'path' => $path,
            'methods' => $methods,
            'code' => $code,
        ];
        if ($description !== null) {
            $params['description'] = $description;
        }
        if ($collection !== null) {
            $params['collection'] = $collection;
        }
        $res = $this->client->sendCommand('create_script', $params);
        return $res['data'] ?? [];
    }

    public function list(): array
    {
        $res = $this->client->sendCommand('list_scripts', ['database' => $this->client->getDatabase()]);
        return $res['data'] ?? [];
    }

    public function get(string $scriptId): array
    {
        $res = $this->client->sendCommand('get_script', [
            'database' => $this->client->getDatabase(),
            'script_id' => $scriptId
        ]);
        return $res['data'] ?? [];
    }

    public function update(string $scriptId, array $updates): array
    {
        $res = $this->client->sendCommand('update_script', [
            'database' => $this->client->getDatabase(),
            'script_id' => $scriptId,
            'updates' => $updates
        ]);
        return $res['data'] ?? [];
    }

    public function delete(string $scriptId): void
    {
        $this->client->sendCommand('delete_script', [
            'database' => $this->client->getDatabase(),
            'script_id' => $scriptId
        ]);
    }

    public function getStats(): array
    {
        $res = $this->client->sendCommand('get_script_stats', []);
        return $res['data'] ?? [];
    }
}

File: clients/Ruby-client/lib/solidb/scripts_client.rb
module SoliDB
  class ScriptsClient
    def initialize(client)
      @client = client
    end

    def create(name:, path:, methods:, code:, description: nil, collection: nil)
      params = {
        database: @client.database,
        name: name,
        path: path,
        methods: methods,
        code: code
      }
      params[:description] = description if description
      params[:collection] = collection if collection
      @client.send_command("create_script", params)
    end

    def list
      @client.send_command("list_scripts", database: @client.database) || []
    end

    def get(script_id)
      @client.send_command("get_script", database: @client.database, script_id: script_id)
    end

    def update(script_id, updates)
      @client.send_command("update_script", database: @client.database, script_id: script_id, updates: updates)
    end

    def delete(script_id)
      @client.send_command("delete_script", database: @client.database, script_id: script_id)
      nil
    end

    def get_stats
      @client.send_command("get_script_stats", {})
    end
  end
end
